[BUG] can't run e2e tests back to back without cleaning up vcsim

**Describe the bug**
I would like to be able to run the e2e tests multiple times without cleaning everything up. Particularly, I don't want to have to recreate the vcsim. 

Perhaps this isn't true, but from what I have seen, vcsim will remember the state of things. So if in the source test, we power off two vms, when we run again and try to power off those same vms, nothing will happen. No events will be sent, and the test will fail.

Same thing happens for the TestBindingGOVC. The tag and category will already exist, so those commands won't prompt any new events.

**To Reproduce**
Steps to reproduce the behavior:

1. setup tests: deploy test/config, create vsphere-credentials
1. Run the e2e tests once `go test -v -race -count=1 -tags=e2e ./test/e2e` (they should pass)
1. Run them again `go test -v -race -count=1 -tags=e2e ./test/e2e` (they should fail)

**Expected behavior**
I want to be able to run `go test -v -race -count=1 -tags=e2e ./test/e2e` multiple times and see the tests pass each time.

**Affected version**
Latest on main

**Screenshots/Debug Output**
Output of failing tests:
```
❯ go test -v -race -count=1 -tags=e2e ./test/e2e   
=== RUN   TestBindingGOVC
    null.go:25: logstream was requested, but SYSTEM_NAMESPACE was unset.
Created binding
    util.go:117:  job <*>&Job{ObjectMeta:{binding-g-o-v-c-pcldfwaw  default  4399e6a2-f015-4606-8ea4-b7dc777198c4 50132912 0 2022-02-04 18:53:28 -0500 EST <nil> <nil> map[job-name:binding-g-o-v-c-ajfgjanq] map[] [] []  [{e2e.test Update batch/v1 2022-02-04 18:53:28 -0500 EST FieldsV1 {"f:metadata":{"f:labels":{".":{},"f:job-name":{}}},"f:spec":{"f:backoffLimit":{},"f:completions":{},"f:parallelism":{},"f:template":{"f:spec":{"f:containers":{"k:{\"name\":\"script\"}":{".":{},"f:args":{},"f:command":{},"f:image":{},"f:imagePullPolicy":{},"f:name":{},"f:resources":{},"f:terminationMessagePath":{},"f:terminationMessagePolicy":{}}},"f:dnsPolicy":{},"f:restartPolicy":{},"f:schedulerName":{},"f:securityContext":{},"f:terminationGracePeriodSeconds":{}}}}}}]},Spec:JobSpec{Parallelism:*1,Completions:*1,ActiveDeadlineSeconds:nil,Selector:&v1.LabelSelector{MatchLabels:map[string]string{controller-uid: 4399e6a2-f015-4606-8ea4-b7dc777198c4,},MatchExpressions:[]LabelSelectorRequirement{},},ManualSelector:nil,Template:{{      0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[controller-uid:4399e6a2-f015-4606-8ea4-b7dc777198c4 job-name:binding-g-o-v-c-pcldfwaw] map[] [] []  []} {[{vsphere-binding {nil nil nil nil nil SecretVolumeSource{SecretName:vsphere-credentials,Items:[]KeyToPath{},DefaultMode:*420,Optional:nil,} nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil}}] [] [{script us.gcr.io/daisy-284300/ck/govc:latest [/bin/bash -c] [export GOVC_URL=$VC_URL
        export GOVC_INSECURE=$VC_INSECURE
        export GOVC_USERNAME=$VC_USERNAME
        export GOVC_PASSWORD=$VC_PASSWORD
        govc tags.category.create testing
        govc tags.create -c testing shrug]  [] [] [{VC_URL https://vcsim.default.svc.cluster.local nil} {VC_INSECURE true nil} {VC_USERNAME  &EnvVarSource{FieldRef:nil,ResourceFieldRef:nil,ConfigMapKeyRef:nil,SecretKeyRef:&SecretKeySelector{LocalObjectReference:LocalObjectReference{Name:vsphere-credentials,},Key:username,Optional:nil,},}} {VC_PASSWORD  &EnvVarSource{FieldRef:nil,ResourceFieldRef:nil,ConfigMapKeyRef:nil,SecretKeyRef:&SecretKeySelector{LocalObjectReference:LocalObjectReference{Name:vsphere-credentials,},Key:password,Optional:nil,},}}] {map[] map[]} [{vsphere-binding true /var/bindings/vsphere  <nil> }] [] nil nil nil nil /dev/termination-log File IfNotPresent nil false false false}] [] Never 0xc00075c128 <nil> ClusterFirst map[]   <nil>  false false false <nil> &PodSecurityContext{SELinuxOptions:nil,RunAsUser:nil,RunAsNonRoot:nil,SupplementalGroups:[],FSGroup:nil,RunAsGroup:nil,Sysctls:[]Sysctl{},WindowsOptions:nil,FSGroupChangePolicy:nil,SeccompProfile:nil,} []   nil default-scheduler [] []  <nil> nil [] <nil> <nil> <nil> map[] [] <nil>}},BackoffLimit:*6,TTLSecondsAfterFinished:nil,CompletionMode:nil,Suspend:nil,},Status:JobStatus{Conditions:[]JobCondition{},StartTime:<nil>,CompletionTime:<nil>,Active:0,Succeeded:0,Failed:0,CompletedIndexes:,},}
    util.go:135: Active=1, Failed=0, Succeeded=0
    util.go:135: Active=1, Failed=0, Succeeded=0
    util.go:135: Active=1, Failed=1, Succeeded=0
    util.go:135: Active=1, Failed=1, Succeeded=0
    util.go:135: Active=1, Failed=1, Succeeded=0
    util.go:135: Active=1, Failed=1, Succeeded=0
    util.go:135: Active=1, Failed=1, Succeeded=0
    util.go:135: Active=1, Failed=1, Succeeded=0
    util.go:135: Active=1, Failed=1, Succeeded=0
    util.go:135: Active=1, Failed=1, Succeeded=0
    util.go:135: Active=1, Failed=1, Succeeded=0
    util.go:135: Active=1, Failed=1, Succeeded=0
    util.go:135: Active=1, Failed=2, Succeeded=0
    util.go:135: Active=1, Failed=2, Succeeded=0
    util.go:135: Active=1, Failed=2, Succeeded=0
    util.go:135: Active=1, Failed=2, Succeeded=0
    util.go:135: Active=1, Failed=2, Succeeded=0
    util.go:135: Active=1, Failed=2, Succeeded=0
    util.go:135: Active=1, Failed=2, Succeeded=0
    util.go:135: Active=1, Failed=2, Succeeded=0
    util.go:135: Active=1, Failed=2, Succeeded=0
    util.go:135: Active=1, Failed=2, Succeeded=0
    util.go:135: Active=1, Failed=2, Succeeded=0
    util.go:135: Active=1, Failed=2, Succeeded=0
    util.go:135: Active=1, Failed=2, Succeeded=0
    util.go:135: Active=1, Failed=2, Succeeded=0
    util.go:135: Active=1, Failed=2, Succeeded=0
    util.go:135: Active=1, Failed=2, Succeeded=0
    util.go:135: Active=1, Failed=2, Succeeded=0
    util.go:135: Active=1, Failed=2, Succeeded=0
    util.go:135: Active=1, Failed=2, Succeeded=0
    util.go:135: Active=1, Failed=2, Succeeded=0
    util.go:135: Active=1, Failed=2, Succeeded=0
    util.go:135: Active=1, Failed=3, Succeeded=0
    util.go:135: Active=1, Failed=3, Succeeded=0
    util.go:135: Active=1, Failed=3, Succeeded=0
    util.go:135: Active=1, Failed=3, Succeeded=0
    util.go:135: Active=1, Failed=3, Succeeded=0
    util.go:135: Active=1, Failed=3, Succeeded=0
    util.go:135: Active=1, Failed=3, Succeeded=0
    util.go:135: Active=1, Failed=3, Succeeded=0
    util.go:135: Active=1, Failed=3, Succeeded=0
    util.go:135: Active=1, Failed=3, Succeeded=0
    util.go:135: Active=1, Failed=3, Succeeded=0
    util.go:135: Active=1, Failed=3, Succeeded=0
    util.go:135: Active=1, Failed=3, Succeeded=0
    util.go:135: Active=1, Failed=3, Succeeded=0
    util.go:135: Active=1, Failed=3, Succeeded=0
    util.go:135: Active=1, Failed=3, Succeeded=0
    util.go:135: Active=1, Failed=3, Succeeded=0
    util.go:135: Active=1, Failed=3, Succeeded=0
    util.go:135: Active=1, Failed=3, Succeeded=0
    util.go:135: Active=1, Failed=3, Succeeded=0
    util.go:135: Active=1, Failed=3, Succeeded=0
    util.go:135: Active=1, Failed=3, Succeeded=0
    util.go:135: Active=1, Failed=3, Succeeded=0
    util.go:135: Active=1, Failed=3, Succeeded=0
    util.go:135: Active=1, Failed=3, Succeeded=0
    util.go:135: Active=1, Failed=3, Succeeded=0
    util.go:135: Active=1, Failed=3, Succeeded=0
    util.go:135: Active=1, Failed=3, Succeeded=0
    util.go:135: Active=1, Failed=3, Succeeded=0
    util.go:141: Error waiting for Job to complete successfully: timed out waiting for the condition
--- FAIL: TestBindingGOVC (60.92s)
=== RUN   TestBindingPowerCLICore
    null.go:25: logstream was requested, but SYSTEM_NAMESPACE was unset.
Created binding
    util.go:117:  job <*>&Job{ObjectMeta:{binding-power-c-l-i-core-tmwkwtjt  default  822a01e8-c8ac-4688-a294-a17dd79ef933 50135691 0 2022-02-04 18:54:30 -0500 EST <nil> <nil> map[job-name:binding-power-c-l-i-core-jcfphdlo] map[] [] []  [{e2e.test Update batch/v1 2022-02-04 18:54:30 -0500 EST FieldsV1 {"f:metadata":{"f:labels":{".":{},"f:job-name":{}}},"f:spec":{"f:backoffLimit":{},"f:completions":{},"f:parallelism":{},"f:template":{"f:spec":{"f:containers":{"k:{\"name\":\"script\"}":{".":{},"f:args":{},"f:command":{},"f:image":{},"f:imagePullPolicy":{},"f:name":{},"f:resources":{},"f:terminationMessagePath":{},"f:terminationMessagePolicy":{}}},"f:dnsPolicy":{},"f:restartPolicy":{},"f:schedulerName":{},"f:securityContext":{},"f:terminationGracePeriodSeconds":{}}}}}}]},Spec:JobSpec{Parallelism:*1,Completions:*1,ActiveDeadlineSeconds:nil,Selector:&v1.LabelSelector{MatchLabels:map[string]string{controller-uid: 822a01e8-c8ac-4688-a294-a17dd79ef933,},MatchExpressions:[]LabelSelectorRequirement{},},ManualSelector:nil,Template:{{      0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[controller-uid:822a01e8-c8ac-4688-a294-a17dd79ef933 job-name:binding-power-c-l-i-core-tmwkwtjt] map[] [] []  []} {[{vsphere-binding {nil nil nil nil nil SecretVolumeSource{SecretName:vsphere-credentials,Items:[]KeyToPath{},DefaultMode:*420,Optional:nil,} nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil}}] [] [{script docker.io/vmware/powerclicore [pwsh -Command] [Set-PowerCLIConfiguration -InvalidCertificateAction Ignore -Confirm:$false | Out-Null
        Connect-VIServer -Server ([System.Uri]$env:VC_URL).Host -User $env:VC_USERNAME -Password $env:VC_PASSWORD
        Get-VIEvent | Write-Host]  [] [] [{VC_URL https://vcsim.default.svc.cluster.local nil} {VC_INSECURE true nil} {VC_USERNAME  &EnvVarSource{FieldRef:nil,ResourceFieldRef:nil,ConfigMapKeyRef:nil,SecretKeyRef:&SecretKeySelector{LocalObjectReference:LocalObjectReference{Name:vsphere-credentials,},Key:username,Optional:nil,},}} {VC_PASSWORD  &EnvVarSource{FieldRef:nil,ResourceFieldRef:nil,ConfigMapKeyRef:nil,SecretKeyRef:&SecretKeySelector{LocalObjectReference:LocalObjectReference{Name:vsphere-credentials,},Key:password,Optional:nil,},}}] {map[] map[]} [{vsphere-binding true /var/bindings/vsphere  <nil> }] [] nil nil nil nil /dev/termination-log File IfNotPresent nil false false false}] [] Never 0xc000472518 <nil> ClusterFirst map[]   <nil>  false false false <nil> &PodSecurityContext{SELinuxOptions:nil,RunAsUser:nil,RunAsNonRoot:nil,SupplementalGroups:[],FSGroup:nil,RunAsGroup:nil,Sysctls:[]Sysctl{},WindowsOptions:nil,FSGroupChangePolicy:nil,SeccompProfile:nil,} []   nil default-scheduler [] []  <nil> nil [] <nil> <nil> <nil> map[] [] <nil>}},BackoffLimit:*6,TTLSecondsAfterFinished:nil,CompletionMode:nil,Suspend:nil,},Status:JobStatus{Conditions:[]JobCondition{},StartTime:<nil>,CompletionTime:<nil>,Active:0,Succeeded:0,Failed:0,CompletedIndexes:,},}
    util.go:135: Active=1, Failed=0, Succeeded=0
    util.go:135: Active=1, Failed=0, Succeeded=0
    util.go:135: Active=1, Failed=0, Succeeded=0
    util.go:135: Active=1, Failed=0, Succeeded=0
    util.go:135: Active=1, Failed=0, Succeeded=0
    util.go:135: Active=1, Failed=0, Succeeded=0
    util.go:135: Active=1, Failed=0, Succeeded=0
    util.go:135: Active=1, Failed=0, Succeeded=0
    util.go:135: Active=1, Failed=0, Succeeded=0
    util.go:135: Active=1, Failed=0, Succeeded=0
    util.go:135: Active=1, Failed=0, Succeeded=0
    util.go:135: Active=0, Failed=0, Succeeded=1
--- PASS: TestBindingPowerCLICore (12.70s)
=== RUN   TestSource
    null.go:25: logstream was requested, but SYSTEM_NAMESPACE was unset.
    util.go:195:  job <*>&Job{ObjectMeta:{source-cmalywuw  default  36df04eb-bac2-4f42-8ed2-639fe88a42ec 50136205 0 2022-02-04 18:54:42 -0500 EST <nil> <nil> map[controller-uid:36df04eb-bac2-4f42-8ed2-639fe88a42ec job-name:source-cmalywuw] map[] [] []  [{e2e.test Update batch/v1 2022-02-04 18:54:42 -0500 EST FieldsV1 {"f:spec":{"f:backoffLimit":{},"f:completions":{},"f:parallelism":{},"f:template":{"f:spec":{"f:containers":{"k:{\"name\":\"listener\"}":{".":{},"f:env":{".":{},"k:{\"name\":\"PORT\"}":{".":{},"f:name":{},"f:value":{}}},"f:image":{},"f:imagePullPolicy":{},"f:name":{},"f:ports":{".":{},"k:{\"containerPort\":8080,\"protocol\":\"TCP\"}":{".":{},"f:containerPort":{},"f:name":{},"f:protocol":{}}},"f:readinessProbe":{".":{},"f:failureThreshold":{},"f:periodSeconds":{},"f:successThreshold":{},"f:tcpSocket":{".":{},"f:port":{}},"f:timeoutSeconds":{}},"f:resources":{},"f:terminationMessagePath":{},"f:terminationMessagePolicy":{}}},"f:dnsPolicy":{},"f:restartPolicy":{},"f:schedulerName":{},"f:securityContext":{},"f:terminationGracePeriodSeconds":{}}}}}}]},Spec:JobSpec{Parallelism:*1,Completions:*1,ActiveDeadlineSeconds:nil,Selector:&v1.LabelSelector{MatchLabels:map[string]string{controller-uid: 36df04eb-bac2-4f42-8ed2-639fe88a42ec,},MatchExpressions:[]LabelSelectorRequirement{},},ManualSelector:nil,Template:{{      0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[controller-uid:36df04eb-bac2-4f42-8ed2-639fe88a42ec job-name:source-cmalywuw] map[] [] []  []} {[] [] [{listener us.gcr.io/daisy-284300/ck/listener:latest [] []  [{http 0 8080 TCP }] [] [{PORT 8080 nil}] {map[] map[]} [] [] nil Probe{Handler:Handler{Exec:nil,HTTPGet:nil,TCPSocket:&TCPSocketAction{Port:{0 8080 },Host:,},},InitialDelaySeconds:0,TimeoutSeconds:1,PeriodSeconds:10,SuccessThreshold:1,FailureThreshold:3,TerminationGracePeriodSeconds:nil,} nil nil /dev/termination-log File IfNotPresent nil false false false}] [] Never 0xc000398588 <nil> ClusterFirst map[]   <nil>  false false false <nil> &PodSecurityContext{SELinuxOptions:nil,RunAsUser:nil,RunAsNonRoot:nil,SupplementalGroups:[],FSGroup:nil,RunAsGroup:nil,Sysctls:[]Sysctl{},WindowsOptions:nil,FSGroupChangePolicy:nil,SeccompProfile:nil,} []   nil default-scheduler [] []  <nil> nil [] <nil> <nil> <nil> map[] [] <nil>}},BackoffLimit:*6,TTLSecondsAfterFinished:nil,CompletionMode:nil,Suspend:nil,},Status:JobStatus{Conditions:[]JobCondition{},StartTime:<nil>,CompletionTime:<nil>,Active:0,Succeeded:0,Failed:0,CompletedIndexes:,},}
Created source
Created binding
    util.go:117:  job <*>&Job{ObjectMeta:{source-kafobbre  default  e62cddf0-6f21-4c7e-9914-972fade6382e 50136415 0 2022-02-04 18:54:45 -0500 EST <nil> <nil> map[job-name:source-fpqvxcjy] map[] [] []  [{e2e.test Update batch/v1 2022-02-04 18:54:45 -0500 EST FieldsV1 {"f:metadata":{"f:labels":{".":{},"f:job-name":{}}},"f:spec":{"f:backoffLimit":{},"f:completions":{},"f:parallelism":{},"f:template":{"f:spec":{"f:containers":{"k:{\"name\":\"script\"}":{".":{},"f:args":{},"f:command":{},"f:image":{},"f:imagePullPolicy":{},"f:name":{},"f:resources":{},"f:terminationMessagePath":{},"f:terminationMessagePolicy":{}}},"f:dnsPolicy":{},"f:restartPolicy":{},"f:schedulerName":{},"f:securityContext":{},"f:terminationGracePeriodSeconds":{}}}}}}]},Spec:JobSpec{Parallelism:*1,Completions:*1,ActiveDeadlineSeconds:nil,Selector:&v1.LabelSelector{MatchLabels:map[string]string{controller-uid: e62cddf0-6f21-4c7e-9914-972fade6382e,},MatchExpressions:[]LabelSelectorRequirement{},},ManualSelector:nil,Template:{{      0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[controller-uid:e62cddf0-6f21-4c7e-9914-972fade6382e job-name:source-kafobbre] map[] [] []  []} {[{vsphere-binding {nil nil nil nil nil SecretVolumeSource{SecretName:vsphere-credentials,Items:[]KeyToPath{},DefaultMode:*420,Optional:nil,} nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil}}] [] [{script us.gcr.io/daisy-284300/ck/govc:latest [/bin/bash -c] [export GOVC_URL=$VC_URL
        export GOVC_INSECURE=$VC_INSECURE
        export GOVC_USERNAME=$VC_USERNAME
        export GOVC_PASSWORD=$VC_PASSWORD
        sleep 5
        govc vm.power -off /DC0/vm/DC0_H0_VM0 && sleep 3
        govc vm.power -off /DC0/vm/DC0_H0_VM1 && sleep 3]  [] [] [{VC_URL https://vcsim.default.svc.cluster.local nil} {VC_INSECURE true nil} {VC_USERNAME  &EnvVarSource{FieldRef:nil,ResourceFieldRef:nil,ConfigMapKeyRef:nil,SecretKeyRef:&SecretKeySelector{LocalObjectReference:LocalObjectReference{Name:vsphere-credentials,},Key:username,Optional:nil,},}} {VC_PASSWORD  &EnvVarSource{FieldRef:nil,ResourceFieldRef:nil,ConfigMapKeyRef:nil,SecretKeyRef:&SecretKeySelector{LocalObjectReference:LocalObjectReference{Name:vsphere-credentials,},Key:password,Optional:nil,},}}] {map[] map[]} [{vsphere-binding true /var/bindings/vsphere  <nil> }] [] nil nil nil nil /dev/termination-log File IfNotPresent nil false false false}] [] Never 0xc0003997d8 <nil> ClusterFirst map[]   <nil>  false false false <nil> &PodSecurityContext{SELinuxOptions:nil,RunAsUser:nil,RunAsNonRoot:nil,SupplementalGroups:[],FSGroup:nil,RunAsGroup:nil,Sysctls:[]Sysctl{},WindowsOptions:nil,FSGroupChangePolicy:nil,SeccompProfile:nil,} []   nil default-scheduler [] []  <nil> nil [] <nil> <nil> <nil> map[] [] <nil>}},BackoffLimit:*6,TTLSecondsAfterFinished:nil,CompletionMode:nil,Suspend:nil,},Status:JobStatus{Conditions:[]JobCondition{},StartTime:<nil>,CompletionTime:<nil>,Active:0,Succeeded:0,Failed:0,CompletedIndexes:,},}
    util.go:135: Active=1, Failed=0, Succeeded=0
    util.go:135: Active=1, Failed=0, Succeeded=0
    util.go:135: Active=1, Failed=0, Succeeded=0
    util.go:135: Active=1, Failed=0, Succeeded=0
    util.go:135: Active=1, Failed=0, Succeeded=0
    util.go:135: Active=1, Failed=0, Succeeded=0
    util.go:135: Active=1, Failed=0, Succeeded=0
    util.go:135: Active=1, Failed=1, Succeeded=0
    util.go:135: Active=1, Failed=1, Succeeded=0
    util.go:135: Active=1, Failed=1, Succeeded=0
    util.go:135: Active=1, Failed=1, Succeeded=0
    util.go:135: Active=1, Failed=1, Succeeded=0
    util.go:135: Active=1, Failed=1, Succeeded=0
    util.go:135: Active=1, Failed=1, Succeeded=0
    util.go:135: Active=1, Failed=1, Succeeded=0
    util.go:135: Active=1, Failed=1, Succeeded=0
    util.go:135: Active=1, Failed=1, Succeeded=0
    util.go:135: Active=1, Failed=2, Succeeded=0
    util.go:135: Active=1, Failed=2, Succeeded=0
    util.go:135: Active=1, Failed=2, Succeeded=0
    util.go:135: Active=1, Failed=2, Succeeded=0
    util.go:135: Active=1, Failed=2, Succeeded=0
    util.go:135: Active=1, Failed=2, Succeeded=0
    util.go:135: Active=1, Failed=2, Succeeded=0
    util.go:135: Active=1, Failed=2, Succeeded=0
    util.go:135: Active=1, Failed=2, Succeeded=0
    util.go:135: Active=1, Failed=2, Succeeded=0
    util.go:135: Active=1, Failed=2, Succeeded=0
    util.go:135: Active=1, Failed=2, Succeeded=0
    util.go:135: Active=1, Failed=2, Succeeded=0
    util.go:135: Active=1, Failed=2, Succeeded=0
    util.go:135: Active=1, Failed=2, Succeeded=0
    util.go:135: Active=1, Failed=2, Succeeded=0
    util.go:135: Active=1, Failed=2, Succeeded=0
    util.go:135: Active=1, Failed=2, Succeeded=0
    util.go:135: Active=1, Failed=2, Succeeded=0
    util.go:135: Active=1, Failed=2, Succeeded=0
    util.go:135: Active=1, Failed=3, Succeeded=0
    util.go:135: Active=1, Failed=3, Succeeded=0
    util.go:135: Active=1, Failed=3, Succeeded=0
    util.go:135: Active=1, Failed=3, Succeeded=0
    util.go:135: Active=1, Failed=3, Succeeded=0
    util.go:135: Active=1, Failed=3, Succeeded=0
    util.go:135: Active=1, Failed=3, Succeeded=0
    util.go:135: Active=1, Failed=3, Succeeded=0
    util.go:135: Active=1, Failed=3, Succeeded=0
    util.go:135: Active=1, Failed=3, Succeeded=0
    util.go:135: Active=1, Failed=3, Succeeded=0
    util.go:135: Active=1, Failed=3, Succeeded=0
    util.go:135: Active=1, Failed=3, Succeeded=0
    util.go:135: Active=1, Failed=3, Succeeded=0
    util.go:135: Active=1, Failed=3, Succeeded=0
    util.go:135: Active=1, Failed=3, Succeeded=0
    util.go:135: Active=1, Failed=3, Succeeded=0
    util.go:135: Active=1, Failed=3, Succeeded=0
    util.go:135: Active=1, Failed=3, Succeeded=0
    util.go:135: Active=1, Failed=3, Succeeded=0
    util.go:135: Active=1, Failed=3, Succeeded=0
    util.go:135: Active=1, Failed=3, Succeeded=0
    util.go:135: Active=1, Failed=3, Succeeded=0
    util.go:135: Active=1, Failed=3, Succeeded=0
    util.go:135: Active=1, Failed=3, Succeeded=0
    util.go:141: Error waiting for Job to complete successfully: timed out waiting for the condition
--- FAIL: TestSource (64.41s)
FAIL
FAIL    github.com/vmware-tanzu/sources-for-knative/test/e2e    138.603s
FAIL
```

**Additional context**
N/A

--- a/.github/workflows/kind-e2e.yaml
+++ b/.github/workflows/kind-e2e.yaml
@@ -59,11 +59,10 @@ jobs:
         # Build and Publish our containers to the docker daemon (including test assets)
         export GO111MODULE=on
         export GOFLAGS=-mod=vendor
-        ko apply -PRf config/ -f test/config
+        ko apply -PRf config/
 
-        # Build Knative plugin and create Secret
+        # Build Knative plugin
         go build -o kn-vsphere ./plugins/vsphere/cmd/vsphere
-        ./kn-vsphere auth create --name vsphere-credentials --username user --password pass
 
         kubectl -n vmware-sources wait --timeout=10s --for=condition=Available deploy/webhook
 

--- a/go.mod
+++ b/go.mod
@@ -36,6 +36,7 @@ require (
 require (
 	github.com/hashicorp/hcl v1.0.0
 	gotest.tools/v3 v3.0.3
+	k8s.io/utils v0.0.0-20211208161948-7d6a63dca704
 	knative.dev/client v0.29.1-0.20220201152332-677276ae0ab3
 )
 
@@ -145,7 +146,6 @@ require (
 	k8s.io/cli-runtime v0.22.5 // indirect
 	k8s.io/gengo v0.0.0-20211129171323-c02415ce4185 // indirect
 	k8s.io/klog/v2 v2.40.1 // indirect
-	k8s.io/utils v0.0.0-20211208161948-7d6a63dca704 // indirect
 	knative.dev/networking v0.0.0-20220131174231-d23a06807e6c // indirect
 	knative.dev/serving v0.29.1-0.20220131194631-770b91180d43 // indirect
 	sigs.k8s.io/kustomize/api v0.8.11 // indirect

--- a/test/e2e/binding_test.go
+++ b/test/e2e/binding_test.go
@@ -26,6 +26,10 @@ func TestBindingGOVC(t *testing.T) {
 
 	clients := test.Setup(t)
 
+	//create vcsim
+	cleanupVcsim := CreateSimulator(t, clients)
+	defer cleanupVcsim()
+
 	selector, cancel := CreateJobBinding(t, clients)
 	defer cancel()
 
@@ -49,6 +53,10 @@ func TestBindingPowerCLICore(t *testing.T) {
 	defer logstream.Start(t)()
 
 	clients := test.Setup(t)
+
+	//create vcsim
+	cleanupVcsim := CreateSimulator(t, clients)
+	defer cleanupVcsim()
 
 	selector, cancel := CreateJobBinding(t, clients)
 	defer cancel()

--- a/test/e2e/source_test.go
+++ b/test/e2e/source_test.go
@@ -25,6 +25,10 @@ func TestSource(t *testing.T) {
 
 	clients := test.Setup(t)
 
+	//create vcsim
+	cleanupVcsim := CreateSimulator(t, clients)
+	defer cleanupVcsim()
+
 	// Create a job/svc that listens for events and then quits N seconds after the first is received.
 	name, wait, cancelListener := RunJobListener(t, clients)
 	defer cancelListener()

--- a/test/e2e/util.go
+++ b/test/e2e/util.go
@@ -292,6 +292,21 @@ func RunJobListener(t *testing.T, clients *test.Clients) (string, context.Cancel
 func CreateSource(t *testing.T, clients *test.Clients, name string) context.CancelFunc {
 	t.Helper()
 
+	//Set a checkpoint in the past in case test creates events before vsphere source is ready
+	checkpointTime := time.Now().Add(time.Minute * -9)
+	checkpointConfigmap, err := clients.KubeClient.CoreV1().ConfigMaps(ns).Create(
+		context.Background(),
+		&corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{Name: fmt.Sprintf("%s-configmap", name)},
+			Data:       map[string]string{"checkpoint": fmt.Sprintf(`{"lastEventKeyTimestamp": "%s"}`, checkpointTime.UTC().Format(time.RFC3339))},
+		},
+		metav1.CreateOptions{},
+	)
+
+	if err != nil {
+		t.Fatalf("Error creating Configmap: %v", err)
+	}
+
 	knativePlugin := root.NewRootCommand(clients.AsPluginClients())
 	knativePlugin.SetArgs([]string{
 		"source",
@@ -304,9 +319,14 @@ func CreateSource(t *testing.T, clients *test.Clients, name string) context.Canc
 		"--sink-api-version", "v1",
 		"--sink-kind", "Service",
 		"--sink-name", name,
+		"--checkpoint-age", "10m",
 	})
 
-	pkgtest.CleanupOnInterrupt(func() { clients.VMWareClient.Sources.Delete(context.Background(), name, metav1.DeleteOptions{}) }, t.Logf)
+	pkgtest.CleanupOnInterrupt(func() {
+		clients.VMWareClient.Sources.Delete(context.Background(), name, metav1.DeleteOptions{})
+		clients.KubeClient.CoreV1().ConfigMaps(ns).Delete(context.Background(), checkpointConfigmap.Name, metav1.DeleteOptions{})
+	}, t.Logf)
+
 	if err := knativePlugin.Execute(); err != nil {
 		t.Fatalf("Error creating source: %v", err)
 	}

--- a/test/e2e/util.go
+++ b/test/e2e/util.go
@@ -35,6 +35,7 @@ const (
 	vsphereCreds = "vsphere-credentials"
 	user         = "user"
 	password     = "password"
+	jobNameKey   = "job-name"
 )
 
 func CreateJobBinding(t *testing.T, clients *test.Clients) (map[string]string, context.CancelFunc) {
@@ -135,6 +136,10 @@ func RunJobScript(t *testing.T, clients *test.Clients, image string, command []s
 		if err != nil {
 			t.Errorf("Error cleaning up Job %s", job.Name)
 		}
+		err = clients.KubeClient.CoreV1().Pods(job.Namespace).DeleteCollection(context.Background(), metav1.DeleteOptions{}, metav1.ListOptions{LabelSelector: fmt.Sprintf("%s=%s", jobNameKey, job.Name)})
+		if err != nil {
+			t.Errorf("Error cleaning up pod for Job %s", job.Name)
+		}
 	}()
 
 	// Wait for the Job to report a successful execution.
@@ -212,6 +217,10 @@ func RunJobListener(t *testing.T, clients *test.Clients) (string, context.Cancel
 		err := clients.KubeClient.BatchV1().Jobs(job.Namespace).Delete(context.Background(), job.Name, metav1.DeleteOptions{})
 		if err != nil {
 			t.Errorf("Error cleaning up Job %s", job.Name)
+		}
+		err = clients.KubeClient.CoreV1().Pods(job.Namespace).DeleteCollection(context.Background(), metav1.DeleteOptions{}, metav1.ListOptions{LabelSelector: fmt.Sprintf("%s=%s", jobNameKey, name)})
+		if err != nil {
+			t.Errorf("Error cleaning up pod for Job %s", job.Name)
 		}
 	}
 	waiter := func() {

--- a/test/e2e/util.go
+++ b/test/e2e/util.go
@@ -33,7 +33,6 @@ const (
 	vcsim        = "vcsim"
 	ns           = "default"
 	vsphereCreds = "vsphere-credentials"
-	mountPath    = "/var/bindings/vsphere"
 	user         = "user"
 	password     = "password"
 )

--- a/test/e2e/util.go
+++ b/test/e2e/util.go
@@ -39,6 +39,7 @@ const (
 )
 
 func CreateJobBinding(t *testing.T, clients *test.Clients) (map[string]string, context.CancelFunc) {
+	ctx := context.Background()
 	t.Helper()
 	name := helpers.ObjectNameForTest(t)
 
@@ -60,14 +61,14 @@ func CreateJobBinding(t *testing.T, clients *test.Clients) (map[string]string, c
 		"--subject-selector", metav1.FormatLabelSelector(&metav1.LabelSelector{MatchLabels: selector}),
 	})
 
-	pkgtest.CleanupOnInterrupt(func() { clients.VMWareClient.Bindings.Delete(context.Background(), name, metav1.DeleteOptions{}) }, t.Logf)
+	pkgtest.CleanupOnInterrupt(func() { clients.VMWareClient.Bindings.Delete(ctx, name, metav1.DeleteOptions{}) }, t.Logf)
 	if err := knativePlugin.Execute(); err != nil {
 		t.Fatalf("Error creating binding: %v", err)
 	}
 
 	// Wait for the Binding to become "Ready"
 	waitErr := wait.PollImmediate(test.PollInterval, test.PollTimeout, func() (bool, error) {
-		b, err := clients.VMWareClient.Bindings.Get(context.Background(), name, metav1.GetOptions{})
+		b, err := clients.VMWareClient.Bindings.Get(ctx, name, metav1.GetOptions{})
 		if apierrs.IsNotFound(err) {
 			return false, nil
 		} else if err != nil {
@@ -82,7 +83,7 @@ func CreateJobBinding(t *testing.T, clients *test.Clients) (map[string]string, c
 	}
 
 	return selector, func() {
-		err := clients.VMWareClient.Bindings.Delete(context.Background(), name, metav1.DeleteOptions{})
+		err := clients.VMWareClient.Bindings.Delete(ctx, name, metav1.DeleteOptions{})
 		if err != nil {
 			t.Errorf("Error cleaning up binding %s", name)
 		}
@@ -98,6 +99,7 @@ func RunPowershellJob(t *testing.T, clients *test.Clients, image, script string,
 }
 
 func RunJobScript(t *testing.T, clients *test.Clients, image string, command []string, script string, selector map[string]string) {
+	ctx := context.Background()
 	job := &batchv1.Job{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      helpers.ObjectNameForTest(t),
@@ -120,9 +122,9 @@ func RunJobScript(t *testing.T, clients *test.Clients, image string, command []s
 		},
 	}
 	pkgtest.CleanupOnInterrupt(func() {
-		clients.KubeClient.BatchV1().Jobs(job.Namespace).Delete(context.Background(), job.Name, metav1.DeleteOptions{})
+		clients.KubeClient.BatchV1().Jobs(job.Namespace).Delete(ctx, job.Name, metav1.DeleteOptions{})
 	}, t.Logf)
-	job, err := clients.KubeClient.BatchV1().Jobs(job.Namespace).Create(context.Background(), job, metav1.CreateOptions{})
+	job, err := clients.KubeClient.BatchV1().Jobs(job.Namespace).Create(ctx, job, metav1.CreateOptions{})
 	if err != nil {
 		t.Fatalf("Error creating Job: %v", err)
 	}
@@ -132,11 +134,11 @@ func RunJobScript(t *testing.T, clients *test.Clients, image string, command []s
 	t.Log("", "job", spew.Sprint(job))
 
 	defer func() {
-		err := clients.KubeClient.BatchV1().Jobs(job.Namespace).Delete(context.Background(), job.Name, metav1.DeleteOptions{})
+		err := clients.KubeClient.BatchV1().Jobs(job.Namespace).Delete(ctx, job.Name, metav1.DeleteOptions{})
 		if err != nil {
 			t.Errorf("Error cleaning up Job %s", job.Name)
 		}
-		err = clients.KubeClient.CoreV1().Pods(job.Namespace).DeleteCollection(context.Background(), metav1.DeleteOptions{}, metav1.ListOptions{LabelSelector: fmt.Sprintf("%s=%s", jobNameKey, job.Name)})
+		err = clients.KubeClient.CoreV1().Pods(job.Namespace).DeleteCollection(ctx, metav1.DeleteOptions{}, metav1.ListOptions{LabelSelector: fmt.Sprintf("%s=%s", jobNameKey, job.Name)})
 		if err != nil {
 			t.Errorf("Error cleaning up pods for Job %s", job.Name)
 		}
@@ -144,7 +146,7 @@ func RunJobScript(t *testing.T, clients *test.Clients, image string, command []s
 
 	// Wait for the Job to report a successful execution.
 	waitErr := wait.PollImmediate(test.PollInterval, test.PollTimeout, func() (bool, error) {
-		js, err := clients.KubeClient.BatchV1().Jobs(job.Namespace).Get(context.Background(), job.Name, metav1.GetOptions{})
+		js, err := clients.KubeClient.BatchV1().Jobs(job.Namespace).Get(ctx, job.Name, metav1.GetOptions{})
 		if apierrs.IsNotFound(err) {
 			return false, nil
 		} else if err != nil {
@@ -162,6 +164,7 @@ func RunJobScript(t *testing.T, clients *test.Clients, image string, command []s
 }
 
 func RunJobListener(t *testing.T, clients *test.Clients) (string, context.CancelFunc, context.CancelFunc) {
+	ctx := context.Background()
 	name := helpers.ObjectNameForTest(t)
 
 	selector := map[string]string{
@@ -202,9 +205,9 @@ func RunJobListener(t *testing.T, clients *test.Clients) (string, context.Cancel
 		},
 	}
 	pkgtest.CleanupOnInterrupt(func() {
-		clients.KubeClient.BatchV1().Jobs(job.Namespace).Delete(context.Background(), job.Name, metav1.DeleteOptions{})
+		clients.KubeClient.BatchV1().Jobs(job.Namespace).Delete(ctx, job.Name, metav1.DeleteOptions{})
 	}, t.Logf)
-	job, err := clients.KubeClient.BatchV1().Jobs(job.Namespace).Create(context.Background(), job, metav1.CreateOptions{})
+	job, err := clients.KubeClient.BatchV1().Jobs(job.Namespace).Create(ctx, job, metav1.CreateOptions{})
 	if err != nil {
 		t.Fatalf("Error creating Job: %v", err)
 	}
@@ -214,11 +217,11 @@ func RunJobListener(t *testing.T, clients *test.Clients) (string, context.Cancel
 	t.Log("", "job", spew.Sprint(job))
 
 	cancel := func() {
-		err := clients.KubeClient.BatchV1().Jobs(job.Namespace).Delete(context.Background(), job.Name, metav1.DeleteOptions{})
+		err := clients.KubeClient.BatchV1().Jobs(job.Namespace).Delete(ctx, job.Name, metav1.DeleteOptions{})
 		if err != nil {
 			t.Errorf("Error cleaning up Job %s", job.Name)
 		}
-		err = clients.KubeClient.CoreV1().Pods(job.Namespace).DeleteCollection(context.Background(), metav1.DeleteOptions{}, metav1.ListOptions{LabelSelector: fmt.Sprintf("%s=%s", jobNameKey, name)})
+		err = clients.KubeClient.CoreV1().Pods(job.Namespace).DeleteCollection(ctx, metav1.DeleteOptions{}, metav1.ListOptions{LabelSelector: fmt.Sprintf("%s=%s", jobNameKey, name)})
 		if err != nil {
 			t.Errorf("Error cleaning up pods for Job %s", job.Name)
 		}
@@ -226,7 +229,7 @@ func RunJobListener(t *testing.T, clients *test.Clients) (string, context.Cancel
 	waiter := func() {
 		// Wait for the Job to report a successful execution.
 		waitErr := wait.PollImmediate(test.PollInterval, test.PollTimeout, func() (bool, error) {
-			js, err := clients.KubeClient.BatchV1().Jobs(test.Namespace).Get(context.Background(), name, metav1.GetOptions{})
+			js, err := clients.KubeClient.BatchV1().Jobs(test.Namespace).Get(ctx, name, metav1.GetOptions{})
 			if apierrs.IsNotFound(err) {
 				t.Logf("Not found: %v", err)
 				return false, nil
@@ -260,9 +263,9 @@ func RunJobListener(t *testing.T, clients *test.Clients) (string, context.Cancel
 		},
 	}
 	pkgtest.CleanupOnInterrupt(func() {
-		clients.KubeClient.CoreV1().Services(svc.Namespace).Delete(context.Background(), svc.Name, metav1.DeleteOptions{})
+		clients.KubeClient.CoreV1().Services(svc.Namespace).Delete(ctx, svc.Name, metav1.DeleteOptions{})
 	}, t.Logf)
-	svc, err = clients.KubeClient.CoreV1().Services(svc.Namespace).Create(context.Background(), svc, metav1.CreateOptions{})
+	svc, err = clients.KubeClient.CoreV1().Services(svc.Namespace).Create(ctx, svc, metav1.CreateOptions{})
 	if err != nil {
 		cancel()
 		t.Fatalf("Error creating Service: %v", err)
@@ -270,7 +273,7 @@ func RunJobListener(t *testing.T, clients *test.Clients) (string, context.Cancel
 
 	// Wait for pods to show up in the Endpoints resource.
 	waitErr := wait.PollImmediate(test.PollInterval, test.PollTimeout, func() (bool, error) {
-		ep, err := clients.KubeClient.CoreV1().Endpoints(svc.Namespace).Get(context.Background(), svc.Name, metav1.GetOptions{})
+		ep, err := clients.KubeClient.CoreV1().Endpoints(svc.Namespace).Get(ctx, svc.Name, metav1.GetOptions{})
 		if apierrs.IsNotFound(err) {
 			return false, nil
 		} else if err != nil {
@@ -289,7 +292,7 @@ func RunJobListener(t *testing.T, clients *test.Clients) (string, context.Cancel
 	}
 
 	return name, waiter, func() {
-		err := clients.KubeClient.CoreV1().Services(svc.Namespace).Delete(context.Background(), svc.Name, metav1.DeleteOptions{})
+		err := clients.KubeClient.CoreV1().Services(svc.Namespace).Delete(ctx, svc.Name, metav1.DeleteOptions{})
 		if err != nil {
 			t.Errorf("Error cleaning up Service %s: %v", svc.Name, err)
 		}
@@ -298,12 +301,13 @@ func RunJobListener(t *testing.T, clients *test.Clients) (string, context.Cancel
 }
 
 func CreateSource(t *testing.T, clients *test.Clients, name string) context.CancelFunc {
+	ctx := context.Background()
 	t.Helper()
 
 	//Set a checkpoint in the past in case test creates events before vsphere source is ready
 	checkpointTime := time.Now().Add(time.Minute * -9)
 	checkpointConfigmap, err := clients.KubeClient.CoreV1().ConfigMaps(ns).Create(
-		context.Background(),
+		ctx,
 		&corev1.ConfigMap{
 			ObjectMeta: metav1.ObjectMeta{Name: fmt.Sprintf("%s-configmap", name)},
 			Data:       map[string]string{"checkpoint": fmt.Sprintf(`{"lastEventKeyTimestamp": "%s"}`, checkpointTime.UTC().Format(time.RFC3339))},
@@ -331,8 +335,8 @@ func CreateSource(t *testing.T, clients *test.Clients, name string) context.Canc
 	})
 
 	pkgtest.CleanupOnInterrupt(func() {
-		clients.VMWareClient.Sources.Delete(context.Background(), name, metav1.DeleteOptions{})
-		clients.KubeClient.CoreV1().ConfigMaps(ns).Delete(context.Background(), checkpointConfigmap.Name, metav1.DeleteOptions{})
+		clients.VMWareClient.Sources.Delete(ctx, name, metav1.DeleteOptions{})
+		clients.KubeClient.CoreV1().ConfigMaps(ns).Delete(ctx, checkpointConfigmap.Name, metav1.DeleteOptions{})
 	}, t.Logf)
 
 	if err := knativePlugin.Execute(); err != nil {
@@ -341,7 +345,7 @@ func CreateSource(t *testing.T, clients *test.Clients, name string) context.Canc
 
 	// Wait for the Source to become "Ready"
 	waitErr := wait.PollImmediate(test.PollInterval, test.PollTimeout, func() (bool, error) {
-		src, err := clients.VMWareClient.Sources.Get(context.Background(), name, metav1.GetOptions{})
+		src, err := clients.VMWareClient.Sources.Get(ctx, name, metav1.GetOptions{})
 		if apierrs.IsNotFound(err) {
 			return false, nil
 		} else if err != nil {
@@ -356,11 +360,11 @@ func CreateSource(t *testing.T, clients *test.Clients, name string) context.Canc
 	}
 
 	return func() {
-		err := clients.VMWareClient.Sources.Delete(context.Background(), name, metav1.DeleteOptions{})
+		err := clients.VMWareClient.Sources.Delete(ctx, name, metav1.DeleteOptions{})
 		if err != nil {
 			t.Errorf("Error cleaning up source %s", name)
 		}
-		err = clients.KubeClient.CoreV1().ConfigMaps(ns).Delete(context.Background(), checkpointConfigmap.Name, metav1.DeleteOptions{})
+		err = clients.KubeClient.CoreV1().ConfigMaps(ns).Delete(ctx, checkpointConfigmap.Name, metav1.DeleteOptions{})
 		if err != nil {
 			t.Errorf("Error cleaning up configmap %s", name)
 		}
@@ -368,44 +372,45 @@ func CreateSource(t *testing.T, clients *test.Clients, name string) context.Canc
 }
 
 func CreateSimulator(t *testing.T, clients *test.Clients) context.CancelFunc {
+	ctx := context.Background()
 	simDeployment, simService := newSimulator(ns)
 	simSecret := newVCSecret(ns, vsphereCreds, user, password)
 
 	pkgtest.CleanupOnInterrupt(func() {
-		clients.KubeClient.AppsV1().Deployments(simDeployment.Namespace).Delete(context.Background(), simDeployment.Name, metav1.DeleteOptions{})
-		clients.KubeClient.CoreV1().Services(simService.Namespace).Delete(context.Background(), simService.Name, metav1.DeleteOptions{})
-		clients.KubeClient.CoreV1().Secrets(simSecret.Namespace).Delete(context.Background(), simSecret.Name, metav1.DeleteOptions{})
+		clients.KubeClient.AppsV1().Deployments(simDeployment.Namespace).Delete(ctx, simDeployment.Name, metav1.DeleteOptions{})
+		clients.KubeClient.CoreV1().Services(simService.Namespace).Delete(ctx, simService.Name, metav1.DeleteOptions{})
+		clients.KubeClient.CoreV1().Secrets(simSecret.Namespace).Delete(ctx, simSecret.Name, metav1.DeleteOptions{})
 	}, t.Logf)
-	secret, err := clients.KubeClient.CoreV1().Secrets(simSecret.Namespace).Create(context.Background(), simSecret, metav1.CreateOptions{})
+	secret, err := clients.KubeClient.CoreV1().Secrets(simSecret.Namespace).Create(ctx, simSecret, metav1.CreateOptions{})
 	if err != nil {
 		t.Fatalf("Error creating Secret: %v", err)
 	}
-	deployment, err := clients.KubeClient.AppsV1().Deployments(simDeployment.Namespace).Create(context.Background(), simDeployment, metav1.CreateOptions{})
+	deployment, err := clients.KubeClient.AppsV1().Deployments(simDeployment.Namespace).Create(ctx, simDeployment, metav1.CreateOptions{})
 	if err != nil {
 		t.Fatalf("Error creating VCSIM Deployment: %v", err)
 	}
-	service, err := clients.KubeClient.CoreV1().Services(simService.Namespace).Create(context.Background(), simService, metav1.CreateOptions{})
+	service, err := clients.KubeClient.CoreV1().Services(simService.Namespace).Create(ctx, simService, metav1.CreateOptions{})
 	if err != nil {
 		t.Fatalf("Error creating VCSIM Service: %v", err)
 	}
 
 	cancel := func() {
-		err := clients.KubeClient.AppsV1().Deployments(deployment.Namespace).Delete(context.Background(), deployment.Name, metav1.DeleteOptions{})
+		err := clients.KubeClient.AppsV1().Deployments(deployment.Namespace).Delete(ctx, deployment.Name, metav1.DeleteOptions{})
 		if err != nil {
 			t.Errorf("Error cleaning up Deployment %s", deployment.Name)
 		}
-		err = clients.KubeClient.CoreV1().Services(service.Namespace).Delete(context.Background(), service.Name, metav1.DeleteOptions{})
+		err = clients.KubeClient.CoreV1().Services(service.Namespace).Delete(ctx, service.Name, metav1.DeleteOptions{})
 		if err != nil {
 			t.Errorf("Error cleaning up Service %s", service.Name)
 		}
-		err = clients.KubeClient.CoreV1().Secrets(secret.Namespace).Delete(context.Background(), secret.Name, metav1.DeleteOptions{})
+		err = clients.KubeClient.CoreV1().Secrets(secret.Namespace).Delete(ctx, secret.Name, metav1.DeleteOptions{})
 		if err != nil {
 			t.Errorf("Error cleaning up Secret %s", secret.Name)
 		}
 	}
 
 	waitErr := wait.PollImmediate(test.PollInterval, test.PollTimeout, func() (bool, error) {
-		depl, err := clients.KubeClient.AppsV1().Deployments(ns).Get(context.Background(), simDeployment.Name, metav1.GetOptions{})
+		depl, err := clients.KubeClient.AppsV1().Deployments(ns).Get(ctx, simDeployment.Name, metav1.GetOptions{})
 		if err != nil {
 			return false, err
 		}

--- a/test/e2e/util.go
+++ b/test/e2e/util.go
@@ -54,7 +54,7 @@ func CreateJobBinding(t *testing.T, clients *test.Clients) (map[string]string, c
 		"--name", name,
 		"--vc-address", "https://vcsim.default.svc.cluster.local",
 		"--skip-tls-verify", "true",
-		"--secret-ref", "vsphere-credentials",
+		"--secret-ref", vsphereCreds,
 		"--subject-api-version", "batch/v1",
 		"--subject-kind", "Job",
 		"--subject-selector", metav1.FormatLabelSelector(&metav1.LabelSelector{MatchLabels: selector}),
@@ -138,7 +138,7 @@ func RunJobScript(t *testing.T, clients *test.Clients, image string, command []s
 		}
 		err = clients.KubeClient.CoreV1().Pods(job.Namespace).DeleteCollection(context.Background(), metav1.DeleteOptions{}, metav1.ListOptions{LabelSelector: fmt.Sprintf("%s=%s", jobNameKey, job.Name)})
 		if err != nil {
-			t.Errorf("Error cleaning up pod for Job %s", job.Name)
+			t.Errorf("Error cleaning up pods for Job %s", job.Name)
 		}
 	}()
 
@@ -220,7 +220,7 @@ func RunJobListener(t *testing.T, clients *test.Clients) (string, context.Cancel
 		}
 		err = clients.KubeClient.CoreV1().Pods(job.Namespace).DeleteCollection(context.Background(), metav1.DeleteOptions{}, metav1.ListOptions{LabelSelector: fmt.Sprintf("%s=%s", jobNameKey, name)})
 		if err != nil {
-			t.Errorf("Error cleaning up pod for Job %s", job.Name)
+			t.Errorf("Error cleaning up pods for Job %s", job.Name)
 		}
 	}
 	waiter := func() {
@@ -323,7 +323,7 @@ func CreateSource(t *testing.T, clients *test.Clients, name string) context.Canc
 		"--name", name,
 		"--vc-address", "https://vcsim.default.svc.cluster.local",
 		"--skip-tls-verify", "true",
-		"--secret-ref", "vsphere-credentials",
+		"--secret-ref", vsphereCreds,
 		"--sink-api-version", "v1",
 		"--sink-kind", "Service",
 		"--sink-name", name,

--- a/test/e2e/util.go
+++ b/test/e2e/util.go
@@ -7,22 +7,35 @@ package e2e
 
 import (
 	"context"
+	"fmt"
 	"testing"
+	"time"
 
 	"github.com/vmware-tanzu/sources-for-knative/plugins/vsphere/pkg/command/root"
 
 	"github.com/davecgh/go-spew/spew"
+	appsv1 "k8s.io/api/apps/v1"
 	batchv1 "k8s.io/api/batch/v1"
 	corev1 "k8s.io/api/core/v1"
 	apierrs "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/intstr"
 	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/utils/pointer"
 	"knative.dev/pkg/apis"
 	pkgtest "knative.dev/pkg/test"
 	"knative.dev/pkg/test/helpers"
 
 	"github.com/vmware-tanzu/sources-for-knative/test"
+)
+
+const (
+	vcsim        = "vcsim"
+	ns           = "default"
+	vsphereCreds = "vsphere-credentials"
+	mountPath    = "/var/bindings/vsphere"
+	user         = "user"
+	password     = "password"
 )
 
 func CreateJobBinding(t *testing.T, clients *test.Clients) (map[string]string, context.CancelFunc) {
@@ -319,5 +332,150 @@ func CreateSource(t *testing.T, clients *test.Clients, name string) context.Canc
 		if err != nil {
 			t.Errorf("Error cleaning up source %s", name)
 		}
+		err = clients.KubeClient.CoreV1().ConfigMaps(ns).Delete(context.Background(), checkpointConfigmap.Name, metav1.DeleteOptions{})
+		if err != nil {
+			t.Errorf("Error cleaning up configmap %s", name)
+		}
+	}
+}
+
+func CreateSimulator(t *testing.T, clients *test.Clients) context.CancelFunc {
+	simDeployment, simService := newSimulator(ns)
+	simSecret := newVCSecret(ns, vsphereCreds, user, password)
+
+	pkgtest.CleanupOnInterrupt(func() {
+		clients.KubeClient.AppsV1().Deployments(simDeployment.Namespace).Delete(context.Background(), simDeployment.Name, metav1.DeleteOptions{})
+		clients.KubeClient.CoreV1().Services(simService.Namespace).Delete(context.Background(), simService.Name, metav1.DeleteOptions{})
+		clients.KubeClient.CoreV1().Secrets(simSecret.Namespace).Delete(context.Background(), simSecret.Name, metav1.DeleteOptions{})
+	}, t.Logf)
+	secret, err := clients.KubeClient.CoreV1().Secrets(simSecret.Namespace).Create(context.Background(), simSecret, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("Error creating Secret: %v", err)
+	}
+	deployment, err := clients.KubeClient.AppsV1().Deployments(simDeployment.Namespace).Create(context.Background(), simDeployment, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("Error creating VCSIM Deployment: %v", err)
+	}
+	service, err := clients.KubeClient.CoreV1().Services(simService.Namespace).Create(context.Background(), simService, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("Error creating VCSIM Service: %v", err)
+	}
+
+	cancel := func() {
+		err := clients.KubeClient.AppsV1().Deployments(deployment.Namespace).Delete(context.Background(), deployment.Name, metav1.DeleteOptions{})
+		if err != nil {
+			t.Errorf("Error cleaning up Deployment %s", deployment.Name)
+		}
+		err = clients.KubeClient.CoreV1().Services(service.Namespace).Delete(context.Background(), service.Name, metav1.DeleteOptions{})
+		if err != nil {
+			t.Errorf("Error cleaning up Service %s", service.Name)
+		}
+		err = clients.KubeClient.CoreV1().Secrets(secret.Namespace).Delete(context.Background(), secret.Name, metav1.DeleteOptions{})
+		if err != nil {
+			t.Errorf("Error cleaning up Secret %s", secret.Name)
+		}
+	}
+
+	waitErr := wait.PollImmediate(test.PollInterval, test.PollTimeout, func() (bool, error) {
+		depl, err := clients.KubeClient.AppsV1().Deployments(ns).Get(context.Background(), simDeployment.Name, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+
+		status := depl.Status
+		for i := range status.Conditions {
+			c := status.Conditions[i]
+			if c.Type == appsv1.DeploymentAvailable && c.Status == corev1.ConditionTrue {
+				return true, nil
+			}
+		}
+		return false, nil
+	})
+	if waitErr != nil {
+		cancel()
+		t.Fatalf("Error waiting for VCSIM deployment to be ready: %v", waitErr)
+	}
+
+	t.Log("vcsim ready")
+
+	return cancel
+}
+
+func newSimulator(namespace string) (*appsv1.Deployment, *corev1.Service) {
+	l := map[string]string{
+		"app": vcsim,
+	}
+
+	sim := appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      vcsim,
+			Namespace: namespace,
+			Labels:    l,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: pointer.Int32(1),
+			Selector: &metav1.LabelSelector{
+				MatchLabels: l,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: l,
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{{
+						Name:  vcsim,
+						Image: "vmware/vcsim:latest",
+						Args: []string{
+							"/vcsim",
+							"-l",
+							":8989",
+						},
+						ImagePullPolicy: corev1.PullIfNotPresent,
+						Ports: []corev1.ContainerPort{
+							{
+								Name:          "https",
+								ContainerPort: 8989,
+							},
+						},
+					}},
+				},
+			},
+		},
+	}
+
+	svc := corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      vcsim,
+			Namespace: namespace,
+			Labels:    l,
+		},
+		Spec: corev1.ServiceSpec{
+			Ports: []corev1.ServicePort{
+				{
+					Name: "https",
+					Port: 443,
+					TargetPort: intstr.IntOrString{
+						IntVal: 8989,
+					},
+				},
+			},
+			Selector: l,
+		},
+	}
+
+	return &sim, &svc
+}
+
+func newVCSecret(namespace, name, username, password string) *corev1.Secret {
+	return &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Data: map[string][]byte{
+			corev1.BasicAuthUsernameKey: []byte(username),
+			corev1.BasicAuthPasswordKey: []byte(password),
+		},
+		Type: corev1.SecretTypeBasicAuth,
 	}
 }

